src/check-signature.ts

```diff
@@ -0,0 +1,46 @@
+import * as path from 'path';
+
+import { spawn } from './spawn';
+import { NotarizeStapleOptions } from './types';
+import debug from 'debug';
+const d = debug('electron-notarize');
+
+const codesignDisplay = async (opts: NotarizeStapleOptions) => {
+  const result = await spawn('codesign', ['-dv', '-vvvv', '--deep', path.basename(opts.appPath)], {
+    cwd: path.dirname(opts.appPath),
+  });
+  return result;
+};
+
+const codesign = async (opts: NotarizeStapleOptions) => {
+  d('attempting to check codesign of app:', opts.appPath);
+  const result = await spawn(
+    'codesign',
+    ['-vvv', '--deep', '--strict', path.basename(opts.appPath)],
+    {
+      cwd: path.dirname(opts.appPath),
+    },
+  );
+
+  return result;
+};
+export async function checkSignatures(opts: NotarizeStapleOptions): Promise<void> {
+  const codesignResult = await codesign(opts);
+  const codesignInfo = await codesignDisplay(opts);
```

Comment:
```suggestion
  const [codesignResult, codesignInfo] = await Promise.all([codesign(opts), codesignDisplay(opts)]);
```

These should be able to be run in parallel, right?

Reply:
yes done